### PR TITLE
Use archive_param_file feature by default

### PR DESCRIPTION
**Pull request by adincebic**
This prevents build failures for targets with large list of files.
**NOTE:** It is already enabled by default in [apple_support](https://github.com/bazelbuild/apple_support) https://github.com/bazelbuild/apple_support/pull/338

### CODE REVIEW

File: xcodeproj/internal/templates/xcodeproj.bazelrc [hunk 1]
```diff
@@ -67,6 +67,10 @@ build:rules_xcodeproj --show_result=0
 build:rules_xcodeproj --noworker_sandboxing
 build:rules_xcodeproj --spawn_strategy=remote,worker,local
 
+# Use param files.
```
Comment (brentleyjones) [inline xcodeproj/internal/templates/xcodeproj.bazelrc:70]: Can you move this up to line 59 (just after the last `--features`. Also add a TODO to remove once apple_support 1.6.0 is our minimum supported version.

Comment (adincebic) [reply xcodeproj/internal/templates/xcodeproj.bazelrc]: Done.

**Review by mattrobmattrob — APPROVED** [review]

**Comment by brentleyjones** [comment]
You'll need to run `./test/update_all_fixtures.sh` to update the fixtures.

**Comment by adincebic** [comment]
> You'll need to run `./test/update_all_fixtures.sh` to update the fixtures.

I am running into this issue, but only on rules_xcodeproj repo. Tried `bazel clean --expunge`, removed output-base folder that I saw referenced in .bazelrc file but still the same.

```
➜  rules_xcodeproj git:(ac/use-archive_param_files-featuree) ✗ ./test/update_all_fixtures.sh

Updating "examples/integration" fixtures

Building expanded release.tar.gz
INFO: Invocation ID: 6e824cfb-5998-44a7-bfa8-670727740e7a
INFO: Streaming build results to: https://app.buildbuddy.io/invocation/6e824cfb-5998-44a7-bfa8-670727740e7a
INFO: Analyzed target //distribution:opt_expanded_release (1 packages loaded, 3 targets configured).
ERROR: /Users/adincebic/developer/rules_xcodeproj/tools/import_indexstores/BUILD:26:14: Compiling Swift module //tools/import_indexstores:import_indexstores.library failed: Exec failed due to IOException: Running '/var/tmp/_bazel_adincebic/install/d6daf49e33736f131b503f89e218c8fe/xcode-locator 15.1.0.15C65' failed with code 1.
This most likely indicates that xcode version 15.1.0.15C65 is not available on the host machine.
Process exited with status 1
stdout: {
	"15.3": "/Applications/Xcode.app/Contents/Developer",
	"15E204a": "/Applications/Xcode.app/Contents/Developer",
	"15.3.0": "/Applications/Xcode.app/Contents/Developer",
	"15": "/Applications/Xcode.app/Contents/Developer",
	"15.3.0.15E204a": "/Applications/Xcode.app/Contents/Developer"
}

stderr: 2024-07-10 17:42:18.283 xcode-locator[1915:30068] Found bundle com.apple.dt.Xcode in file:///Applications/Xcode.app/; contents on disk: (
    "file:///Applications/Xcode.app/Contents/"
)
2024-07-10 17:42:18.284 xcode-locator[1915:30068] Version strings for file:///Applications/Xcode.app/: short=15.3, expanded=15.3.0
Target //distribution:opt_expanded_release failed to build
Use --verbose_failures to see the command lines of failed build steps.
ERROR: /Users/adincebic/developer/rules_xcodeproj/tools/import_indexstores/BUILD:26:14 Linking tools/import_indexstores/libimport_indexstores.library.a failed: Exec failed due to IOException: Running '/var/tmp/_bazel_adincebic/install/d6daf49e33736f131b503f89e218c8fe/xcode-locator 15.1.0.15C65' failed with code 1.
This most likely indicates that xcode version 15.1.0.15C65 is not available on the host machine.
Process exited with status 1
stdout: {
	"15.3": "/Applications/Xcode.app/Contents/Developer",
	"15E204a": "/Applications/Xcode.app/Contents/Developer",
	"15.3.0": "/Applications/Xcode.app/Contents/Developer",
	"15": "/Applications/Xcode.app/Contents/Developer",
	"15.3.0.15E204a": "/Applications/Xcode.app/Contents/Developer"
}

stderr: 2024-07-10 17:42:18.283 xcode-locator[1915:30068] Found bundle com.apple.dt.Xcode in file:///Applications/Xcode.app/; contents on disk: (
    "file:///Applications/Xcode.app/Contents/"
)
2024-07-10 17:42:18.284 xcode-locator[1915:30068] Version strings for file:///Applications/Xcode.app/: short=15.3, expanded=15.3.0
INFO: Elapsed time: 0.529s, Critical Path: 0.40s
INFO: 3 processes: 3 internal.
ERROR: Build did NOT complete successfully
INFO: Streaming build results to: https://app.buildbuddy.io/invocation/6e824cfb-5998-44a7-bfa8-670727740e7a
```

**Comment by adincebic** [comment]
@brentleyjones I updated the fixtures.

**Pull request closed without merging**